cdap-docs/tools/docs-search-deprecated.py

```diff
@@ -42,91 +43,150 @@ def parse_options():
     """
 
     parser = OptionParser(
-        usage="%prog [options]",
-        description="Searches for deprecated items in documentation ('cdap-docs') and examples ('cdap-examples')")
-
-    parser.add_option(
-        "-r", "--release",
-        dest="release",
-        help="The release to be checked, such as '3.6.0'; default is 'current'",
-        default='current')
+        usage="%prog [release]",
+        description="Searches for deprecated items in documentation ('cdap-docs') and examples ('cdap-examples'). "
+            "If no release is specified, 'current' is used instead. "
+            "Uses the lists at 'http://docs.cask.co/cdap/[release]/en/reference-manual/javadocs/deprecated-list.html' "
+            "and at 'http://docs.cask.co/cdap/json-versions.js'")
 
     (options, args) = parser.parse_args()
 
     return options, args, parser
 
+def load_json_versions(release):
+    json_versions_url = 'http://docs.cask.co/cdap/json-versions.js'
+    page = urllib2.urlopen(json_versions_url).read().strip()
+    if page.startswith(r'versionscallback('):
+        page = page[len(r'versionscallback('):]
+    if page.endswith(r');'):
+        page = page[:-len(r');')]
+    d = ast.literal_eval(page)
+    versions = []
+    for timeline in d['timeline']:
+        versions.append(timeline[1])
+    versions.sort()
+    if release == 'current':
+        versions.append('current')
+    elif release.endswith('SNAPSHOT'):
+        versions.append('current')
+        versions.append(release)
+    elif release in versions:
+        versions = versions[:versions.index(release)+1]
+    return versions
 
 def load_deprecated_items(release):
-    deprecated_url = "http://docs.cask.co/cdap/%s/en/reference-manual/javadocs/deprecated-list.html" % release
+    versions = load_json_versions(release)
+    deprecated_items = dict()
+    longest = 0
+    for version in versions:
+        deprecated_url = "http://docs.cask.co/cdap/%s/en/reference-manual/javadocs/deprecated-list.html" % version
+        deprecated_version = _load_deprecated_items(deprecated_url)
+        for deprecated in deprecated_version:
+            signatures = deprecated_version[deprecated]
+            if deprecated not in deprecated_items:
+                deprecated_items[deprecated] = []
+            for signature in signatures:
+                if signature not in deprecated_items[deprecated]:
+                    deprecated_items[deprecated].append(signature)
+            if len(deprecated) > longest:
+                longest = len(deprecated)
+    return deprecated_items, longest
+
+def _load_deprecated_items(deprecated_url):
+
     print "Loading deprecated info from '%s'" % deprecated_url
     page = urllib2.urlopen(deprecated_url).read()
     soup = BeautifulSoup(page, 'html.parser')
-    soup.prettify()
+#     soup.prettify()
```

Comment:
Actually changed my mind and removed it as you suggested; it wasn't what I was thinking of.